src/proof/nova.rs

```diff
@@ -1472,72 +1472,70 @@ mod tests {
 
     #[test]
     #[ignore]
-    fn outer_prove_evaluate_zero_arg_lambda1() {
+    fn outer_prove_evaluate_zero_arg_lambda() {
         let s = &mut Store::<Fr>::default();
         let expected = s.num(123);
         let terminal = s.get_cont_terminal();
         nova_test_aux(
             s,
-            "((lambda () 123))",
+            "((lambda(x) x) 123)",
             Some(expected),
             None,
             Some(terminal),
             None,
-            3,
+            4,
         );
     }
 
     #[test]
     #[ignore]
     fn outer_prove_evaluate_zero_arg_lambda2() {
         let s = &mut Store::<Fr>::default();
-        let expected = s.num(10);
+        let expected = s.num(123);
         let terminal = s.get_cont_terminal();
         nova_test_aux(
             s,
-            "(let ((x 9) (f (lambda () (+ x 1)))) (f))",
+            "((lambda (y) ((lambda (x) y) 321)) 123)",
             Some(expected),
             None,
             Some(terminal),
             None,
-            10,
+            9,
         );
     }
+
     #[test]
-    fn outer_prove_evaluate_zero_arg_lambda3() {
+    fn outer_prove_evaluate_lambda3() {
         let s = &mut Store::<Fr>::default();
-        let expected = {
-            let arg = s.sym("x");
-            let num = s.num(123);
-            let body = s.list(&[num]);
-            let env = s.nil();
-            s.intern_fun(arg, body, env)
-        };
+        let expr = "((lambda (y) ((lambda (x) ((lambda (z) z) x)) y)) 123)";
+
+        let expected = s.num(123);
         let terminal = s.get_cont_terminal();
-        nova_test_aux(
-            s,
-            "((lambda (x) 123))",
-            Some(expected),
-            None,
-            Some(terminal),
-            None,
-            3,
-        );
+        nova_test_aux(s, expr, Some(expected), None, Some(terminal), None, 10);
     }
 
     #[test]
-    fn outer_prove_evaluate_zero_arg_lambda4() {
+    fn outer_prove_evaluate_lambda4() {
         let s = &mut Store::<Fr>::default();
-        let error = s.get_cont_error();
-        nova_test_aux(s, "((lambda () 123) 1)", None, None, Some(error), None, 3);
+        let expr =
+            // NOTE: We pass two different values. This tests which is returned.
+            "((lambda (y) ((lambda (x) ((lambda (z) z) x)) 888)) 999)";
+
+        let expected = s.num(888);
+        let terminal = s.get_cont_terminal();
+        nova_test_aux(s, expr, Some(expected), None, Some(terminal), None, 10);
```

Comment:
As above, and at the risk of repetition. The only similarity between these tests seems to be the suffix `4`. I think you've tried to reconcile two distinct groups of tests, one for zero-arg lambdas and one for general lambdas whose only similarity is that they are numbered. You could accomplish that by just adding the non-zero-arg tests you intend, without renaming or replacing the zero-arg versions.